test/test_util.jl

```diff
@@ -39,17 +51,24 @@ function test_scalar(f, x; rtol=1e-9, atol=1e-9, fdm=_fdm, test_wirtinger=x isa
             ∂ = 0.5(∂Re - im*∂Im)
             ∂̅ = 0.5(∂Re + im*∂Im)
             @test isapprox(
-                wirtinger_primal(∂x(1)), ∂;
+                wirtinger_primal(∂x), ∂;
                 rtol=rtol, atol=atol, kwargs...
             )
             @test isapprox(
-                wirtinger_conjugate(∂x(1)), ∂̅;
+                wirtinger_conjugate(∂x), ∂̅;
                 rtol=rtol, atol=atol, kwargs...
             )
         end
     end
 end
 
+function ensure_not_running_on_functor(f, name)
```

Comment:
this is temporary and will change in future? let's comment that?

Reply:
Not super soon,
 `@test_scalar` would need a lot of work to handle functors.

Reply:
yeah, i'm fine with that... maybe we should have an issue tracking that and link to it here?

it certainly is on the long-term plan (although someone might need to want it enough for it to actually happen)